spec/operations/story_operations_spec.rb

```diff
@@ -127,9 +145,29 @@
       story.save!
     end
 
-    subject do
-      lambda do
-        StoryOperations::Update.call(story, { state: 'accepted', accepted_at: Date.current }, user)
+    let(:params) { { state: 'accepted', accepted_at: Date.current } }
+
+    subject { -> { StoryOperations::Update.call(story, params, user) } }
+
+    Story::ESTIMABLE_STORY_TYPES.each do |story_type|
+      context "when estimate #{story_type} story" do
+        let(:params) { { 'estimate' => 1, 'story_type' => story_type, 'state' => 'unscheduled' } }
+
+        it 'sets the story state as unstarted' do
+          story.state = 'unscheduled'
+          subject.call
+          expect(story.state).to eq('unstarted')
+        end
+      end
+
+      context "when underestimate #{story_type} story" do
+        let(:params) { { 'estimate' => nil, 'story_type' => story_type } }
+
+        it 'sets the story state as unscheduled' do
+          story.attributes = { estimate: 1, state: 'unstarted' }
+          subject.call
+          expect(story.state).to eq('unscheduled')
+        end
```

Comment:
Can we add a test for the following scenario:

- we have a _started_ estimable story with an estimate of 1
- we change only the estimate of this story to 2
- we verify it didn't affect the state, it remains _starter_

Reply:
Sure!